postgres/osfamilymap.yaml

```diff
@@ -61,7 +61,11 @@ RedHat:
     humanname: PostgreSQL {{ repo.version }} $releasever - $basearch
     gpgcheck: 1
     gpgkey: 'https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG-{{ release }}'
+{%- if grains.get('osfullname', "") == "Amazon Linux" and grains.get('osmajorrelease', 0) == 2 %}
+    baseurl: 'https://download.postgresql.org/pub/repos/yum/{{ repo.version }}/redhat/rhel-7-$basearch'
+{%- else %}
     baseurl: 'https://download.postgresql.org/pub/repos/yum/{{ repo.version }}/redhat/rhel-$releasever-$basearch'
+{%- endif %}
```

Comment:
@Niyakiy This `if` block can be avoided by adding this change to `osmap.yaml` instead, in a similar manner to this block:

https://github.com/saltstack-formulas/postgres-formula/blob/43c56966e784995248808d48d38eee6b8af00b3a/postgres/osmap.yaml#L6-L9

I.e. Probably:

```yaml
Amazon:
  pkg_repo:
    baseurl: 'https://download.postgresql.org/pub/repos/yum/{{ repo.version }}/redhat/rhel-7-$basearch'
```

This isn't _perfect_, since it doesn't specify the version of Amazon Linux but it is probably sufficient, since we're no longer supporting Amazon Linux 1 (2018).  We _could_ use the `oscodename` grain (`Amazon Linux 2`) in `codenamemap.yaml` but that's overkill in my opinion.  Ultimately, we have a new `map.jinja` that is being introduced across formulas and we can provide the best solution when that becomes available.

Reply:
@myii Thanks for the heads up on a better and easier way to introduce a fix. Having AL1 already deprecated it looks reasonable to leave `Amazon` inside `osmap.yaml` as we have only the 2nd version available. However, it will require additional code changes here once AWS releases a new major version of AL. 
I'm about to push the requested code changes along with reworded commit message.